libraries/AP_IOMCU/iofirmware/iofirmware.cpp

```diff
@@ -73,9 +103,17 @@ static void setup_rx_dma(hal_uart_driver* uart)
     dmaStreamDisable(uart->dmarx);
     dmaStreamSetMemory0(uart->dmarx, &iomcu.rx_io_packet);
     dmaStreamSetTransactionSize(uart->dmarx, sizeof(iomcu.rx_io_packet));
+#if defined(STM32H7)
+    dmaStreamSetPeripheral(uart->dmarx, &(uart->usart->RDR));
+#else
     dmaStreamSetPeripheral(uart->dmarx, &(uart->usart->DR));
+#endif
     dmaStreamSetMode(uart->dmarx, uart->dmarxmode    | STM32_DMA_CR_DIR_P2M |
-                     STM32_DMA_CR_MINC | STM32_DMA_CR_TCIE);
+                     STM32_DMA_CR_MINC | STM32_DMA_CR_TCIE
+#if defined(STM32H7)
+                     | (1<<20)
```

Comment:
needs a define

Reply:
fixed through define and also in UARTDriver